core/field_checkbox.js

```diff
@@ -116,6 +131,19 @@ Blockly.FieldCheckbox.prototype.CURSOR = 'default';
  */
 Blockly.FieldCheckbox.prototype.isDirty_ = false;
 
+/**
+ * Configure the field based on the given map of options.
+ * @param {Object} opt_config A map of options to configure the field based on.
+ * @private
+ */
+Blockly.FieldCheckbox.prototype.configure_ = function(opt_config) {
+  if (opt_config) {
```

Comment:
As I think about this more. Perhaps the configure method should be public. 
This allows a user to either pass the property bag in the constructor, or call it at any point after.